src/app/frontend/deploy/deploy_label_service.js

```diff
@@ -0,0 +1,160 @@
+  }
+
+  /**
+   * Deletes row from labels list.
+   * @param {!Array<!DeployFormLabel>} labelList
+   * @param {!DeployFormLabel} label
+   */
+  deleteLabel(labelList, label) {
+    for (let i = 0; i < labelList.length; i++) {
+      if (labelList[i] === label) {
+        labelList.splice(i, 1);
+        break;
+      }
+    }
+  }
+
+  /**
+   * Binds name param to non-editable app label in order to
+   * autocomplete 'app' value label when name value is filled.
+   * @param {!Array<!DeployFormLabel>} labelList
+   * @param {string} name
+   */
+  bindNameLabel(labelList, name) {
+    labelList.filter(function (label) {
+      return label.key === APP_LABEL_KEY;
+    })[0].value = name;
+  }
+
+  /**
+   * Binds containerImage param to non-editable version label in order to
+   * autocomplete 'version' label value when container image value is filled.
+   * @param {!Array<!DeployFormLabel>} labelList
+   * @param {string} containerImage
+   */
+  bindVersionLabel(labelList, containerImage) {
+    /** @type {number} */
+    let index = containerImage.lastIndexOf(':');
+
+    /** @type {string} */
+    let version = '';
+
+    if (index > -1) {
+      version = containerImage.substring(index + 1);
+    }
+
+    labelList.filter(function (label) {
+      return label.key === VERSION_LABEL_KEY;
+    })[0].value = version;
+  }
+
+  /**
+   * Returns true when label is editable and is not last on the list.
+   * Used to indicate whether delete icon should be shown near label.
+   * @param {!Array<!DeployFormLabel>} labelList
+   * @param {!DeployFormLabel} label
+   * @returns {boolean}
+   */
+  isRemovable(labelList, label) {
+    /** @type {!DeployFormLabel} */
+    let lastElement = labelList[labelList.length - 1];
+    return !!(label.editable && label !== lastElement && label.hovered);
+  }
+
+  /**
+   * Shows delete icon after label definition when mouse enters the list item.
+   * @param {!DeployFormLabel} label
+   */
+  onEnter(label) {
+    label.hovered = true;
+  }
+
+  /**
+   * Hides delete icon after label definition when mouse leaves the list item.
+   * @param {!DeployFormLabel} label
+   */
+  onLeave(label) {
+    label.hovered = false;
+  }
+
+  /**
+   * Returns true if there are 2 or more labels with the same key on the labelList,
+   * false otherwise.
+   * @param {!Array<!DeployFormLabel>} labelList
+   * @param {!DeployFormLabel} label
+   * @returns {boolean}
+   */
+  isDuplicated(labelList, label) {
+    /** @type {number} */
+    let duplications = 0;
+    for (let i = 0; i < labelList.length; i++) {
+      if (label.key.length !== 0 && labelList[i].key === label.key) {
+        duplications++;
+      }
+    }
+
+    return duplications > 1;
+  }
+
+  /**
+   * Returns true if there is at least 1 label with 'hasError' property set to true.
+   * @param {!Array<!DeployFormLabel>} labelList
+   * @returns {boolean}
+   */
+  hasDuplicationError(labelList) {
+    for (let i = 0; i < labelList.length; i++) {
```

Comment:
Use Array.prototype.some or Array.protorype.every for such things. Such code is more idiomatic and easier to read.